src/DevMenu/ui/SaveFileDev.tsx

```diff
@@ -6,48 +6,87 @@ import AccordionDetails from "@mui/material/AccordionDetails";
 import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
 
 import Typography from "@mui/material/Typography";
-import { saveObject } from "../../SaveObject";
+import { loadGame, saveObject } from "../../SaveObject";
 import { SnackbarEvents } from "../../ui/React/Snackbar";
 import { ToastVariant } from "@enums";
 import { Upload } from "@mui/icons-material";
 import { Button } from "@mui/material";
 import { OptionSwitch } from "../../ui/React/OptionSwitch";
-import { isBinaryFormat } from "../../../electron/saveDataBinaryFormat";
+import { SaveData } from "../../types";
+import { GetAllServers, loadAllServers, saveAllServers } from "../../Server/AllServers";
+import { Player } from "../../Player";
+import { killWorkerScriptByPid } from "../../Netscript/killWorkerScript";
+import { loadAllRunningScripts } from "../../NetscriptWorker";
 
 export function SaveFileDev(): React.ReactElement {
   const importInput = useRef<HTMLInputElement>(null);
-  const [saveFile, setSaveFile] = useState("");
+  const [saveData, setSaveData] = useState<SaveData>();
   const [restoreScripts, setRestoreScripts] = useState(true);
-  const [restoreAugs, setRestoreAugs] = useState(true);
+  const [restoreAugmentations, setRestoreAugmentations] = useState(true);
   const [restoreSFs, setRestoreSFs] = useState(true);
 
   async function onImport(event: React.ChangeEvent<HTMLInputElement>): Promise<void> {
     try {
       const saveData = await saveObject.getSaveDataFromFile(event.target.files);
-      // TODO Support binary format. This is low priority because this entire feature (SaveFileDev) is not fully
-      // implemented. "doRestore" does nothing.
-      if (isBinaryFormat(saveData)) {
-        SnackbarEvents.emit("We currently do not support binary format", ToastVariant.ERROR, 5000);
-        return;
-      }
-      const save = decodeURIComponent(escape(atob(saveData as string)));
-      setSaveFile(save);
+      setSaveData(saveData);
     } catch (e: unknown) {
       SnackbarEvents.emit(String(e), ToastVariant.ERROR, 5000);
     }
   }
 
   function startImport(): void {
-    if (!window.File || !window.FileReader || !window.FileList || !window.Blob) return;
-    const ii = importInput.current;
-    if (ii === null) throw new Error("import input should not be null");
-    ii.click();
+    if (!window.File || !window.FileReader || !window.FileList || !window.Blob) {
+      return;
+    }
+    if (!importInput.current) {
+      throw new Error("Invalid import input");
+    }
+    importInput.current.click();
   }
 
-  function doRestore(): void {
-    const save = JSON.parse(saveFile);
-    // TODO unplanned: "Continue here". Unknown what this todo is for.
-    console.error(save);
+  async function doRestore(): Promise<void> {
```

Comment:
Does this do anything different from regular import? Is there a reason to keep it in the dev menu?

Reply:
I only implement the needed changes to make this "feature" work.

To be honest, I have never used this "SaveFileDev" and I think its usuage is incredibly niche:
- Import without reloading.
- Can exclude other things while importing (scripts, augmentations, SFs).

I'm fine if you want to remove this "SaveFileDev" feature. I use my own tools to save/load while testing.

Reply:
For context, this feature was added in 8159dad5fa63ed11b4320d583015060a57fd819b, but the implementation have been incomplete until now. I think that nobody uses this feature, so nobody want to finish the implementation. It may be fine if we just remove it.

Reply:
I've also never used it when testing. @Snarling?

Reply:
Yeah I've never used it. I agree we should probably just remove it